src/radix_connect/mobile/relay_service/models/request/request.rs

```diff
@@ -0,0 +1,306 @@
+    fn send_request() {
+        let session_id = SessionID::sample();
+        let data = BagOfBytes::sample();
+        let request = SUT::new_send_request(session_id.clone(), data.clone());
+        assert_eq!(request.method, Method::SendRequest);
+        assert_eq!(request.session_id, session_id);
+        assert_eq!(request.data, Some(data));
+    }
+
+    #[test]
+    fn get_requests() {
+        let session_id = SessionID::sample();
+        let request = SUT::new_get_requests(session_id.clone());
+        assert_eq!(request.method, Method::GetRequests);
+        assert_eq!(request.session_id, session_id);
+        assert_eq!(request.data, None);
+    }
+
+    #[test]
+    fn send_response() {
+        let session_id = SessionID::sample();
+        let data = BagOfBytes::sample();
+        let request = SUT::new_send_response(session_id.clone(), data.clone());
+        assert_eq!(request.method, Method::SendResponse);
+        assert_eq!(request.session_id, session_id);
+        assert_eq!(request.data, Some(data));
+    }
+
+    #[test]
+    fn get_responses() {
+        let session_id = SessionID::sample();
+        let request = SUT::new_get_responses(session_id.clone());
+        assert_eq!(request.method, Method::GetResponses);
+        assert_eq!(request.session_id, session_id);
+        assert_eq!(request.data, None);
+    }
+
+    #[test]
+    fn get_handshake_requests() {
+        let session_id = SessionID::sample();
+        let request = SUT::new_get_handshake_request(session_id.clone());
+        assert_eq!(request.method, Method::GetHandshakeRequest);
+        assert_eq!(request.session_id, session_id);
+        assert_eq!(request.data, None);
+    }
+
+    #[test]
+    fn send_handshake_response() {
+        let session_id = SessionID::sample();
+        let data = BagOfBytes::sample();
+        let request =
+            SUT::new_send_handshake_response(session_id.clone(), data.clone());
+        assert_eq!(request.method, Method::SendHandshakeResponse);
+        assert_eq!(request.session_id, session_id);
+        assert_eq!(request.data, Some(data));
+    }
+
+    #[test]
+    fn send_handshake_response_with_public_key() {
+        let session_id = SessionID::sample();
+        let public_key = PublicKey::sample();
+        let request = SUT::new_send_handshake_response_with_public_key(
+            session_id.clone(),
+            public_key.clone(),
+        );
+
+        assert_eq!(request.method, Method::SendHandshakeResponse);
+        assert_eq!(request.session_id, session_id);
+        assert_eq!(
+            request.data,
+            Some(BagOfBytes::from_hex(public_key.to_hex().as_str()).unwrap())
+        );
+    }
+
+    #[test]
+    fn send_request_json_roundtrip() {
+        let session_id = SessionID::sample();
+        let data = BagOfBytes::sample();
+        let request = SUT::new_send_request(session_id.clone(), data.clone());
+
+        let expected_json = format!(
+            r#"
+            {{
```

Comment:
nice usage of format! and `{{` :)

Reply:
ok I changed my mind, lets don't do this. This actually defeats the purpose right? We are no longer asserting against something "recorded", so we should not use format, rather put explicit values. Otherwise we are not actually asserting that an expected to pass JSON value will successfully deserialize right?!